cmd/snap/cmd_model.go

```diff
@@ -125,19 +144,77 @@ func (x *cmdModel) Execute(args []string) error {
 
 	w := tabWriter()
 
+	if x.Serial && client.IsAssertionNotFoundError(serialErr) {
+		// for serial assertion, the primary keys are output (model and
+		// brand-id), but if we didn't find the serial assertion then we still
+		// output the brand-id and model from the model assertion, but also
+		// return a devNotReady error
+		fmt.Fprintf(w, "brand-id:\t%s\n", modelAssertion.HeaderString("brand-id"))
+		fmt.Fprintf(w, "model:\t%s\n", modelAssertion.HeaderString("model"))
+
+		return errors.New(devNotReadyMessage)
```

Comment:
device not registered yet (no serial assertion found) (see forum topic)

Reply:
okay fixed